lib/private/assistant-client.ts

```
@@ -29,35 +29,40 @@ export class AssistantClientImp implements AssistantClient {
 	async search(req: SearchRequest): Promise<SearchResponse> {
 		return this.http.post('/search', { ...req, assistantId: this.id });
 	}
-	async chat(req: ChatRequest): Promise<ChatResponse> {
+	async *chat(req: ChatRequest): AsyncGenerator<ChatResponseType> {
 		const res = this.http.fetchAndStreamData('/chat', { ...req, assistantId: this.id });
-		const decoder = new TextDecoder();
-		let current = '';
+		if (req.stream) {
+			const decoder = new TextDecoder();
+			let current = '';
 
-		for await (const chunk of res) {
-			const buffer = new TextEncoder().encode(chunk);
-			const decodedChunk = decoder.decode(buffer, { stream: true });
-			const lines = decodedChunk.split('\n');
-
-			for (const line of lines.filter((l) => l.length > 0)) {
-				if (line.startsWith('data: ')) {
-					current = line.slice(6);
-				} else {
-					current += line;
+			for await (const chunk of res) {
+				const uint8Array = new TextEncoder().encode(chunk);
+				const decodedChunk = decoder.decode(uint8Array, { stream: true });
+				const lines = decodedChunk.split('\n');
+				for (const line of lines.filter((l) => l.length > 0)) {
+					if (line.startsWith('data: ')) {
+						current = line.slice(6);
+						try {
+							const response: ChatResponseType = JSON.parse(current);
+							yield response;
+							current = '';
+						} catch (error) {
+							console.error('Error parsing response:', error);
```

no need for the log, or the error. just let it propagate to the calling method